src/frontend/packages/lib_video/src/components/common/VideoWidgetProvider/widgets/WidgetThumbnail/index.tsx

```diff
@@ -59,6 +74,37 @@ export const WidgetThumbnail = ({ isLive = true }: WidgetThumbnailProps) => {
     thumbnail: state.getThumbnail(),
   }));
   const hiddenFileInputRef = React.useRef<Nullable<HTMLInputElement>>(null);
+  const { isLoading, data } = useThumbnailMetadata(intl.locale);
+
+  const handleChange = useCallback(
+    async (event: React.ChangeEvent<HTMLInputElement>) => {
+      if (event.target.files && event.target.files[0]) {
+        let thumbnailId;
+        try {
+          if (!thumbnail) {
+            const response = await createThumbnail(event.target.files[0].size);
+            addThumbnail(response);
+            thumbnailId = response.id;
+          } else {
+            thumbnailId = thumbnail.id;
+          }
+          addUpload(modelName.THUMBNAILS, thumbnailId, event.target.files[0]);
+        } catch (error) {
+          if ((error as object).hasOwnProperty('size') && data) {
+            toast.error(
+              intl.formatMessage(messages.errorFileTooLarge, {
+                uploadMaxSize: formatSizeErrorScale(data.upload_max_size_bytes),
+              }),
+            );
+          } else {
+            report(error);
+            toast.error(intl.formatMessage(messages.errorFileUpload));
+          }
+        }
+      }
+    },
+    [addThumbnail, addUpload, intl, data, thumbnail],
```

Comment:
```suggestion
    [addThumbnail, addUpload, intl, data?.upload_max_size_bytes, thumbnail?.id],
```

Reply:
I can't be this precise on thumbnail because I use the whole object.

Reply:
Yes you could, too late it is merged.. it is for this kind of mistake that your test could be complicated, by being not specific enough your useCallback keep refreshing.